conformance/utils/kubernetes/helpers_test.go

```diff
@@ -0,0 +1,160 @@
+package kubernetes
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/gateway-api/apis/v1alpha2"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+func TestSupportedKindsMatch(t *testing.T) {
+	tests := []struct {
```

Comment:
Somehow I had a comment here that got dropped from my previous review.

While looking at coverage for this, there are a few cases we're not covering:

```golang
        if len(expected) != len(actual) {
                t.Logf("Expected %d Gateway status listeners, got %d", len(expected), len(actual))
                return false
        }
```

```golang
                if aListener.Name != eListener.Name {
                        t.Logf("Name doesn't match")
                        return false
                }
```

```golang
                if aListener.AttachedRoutes != eListener.AttachedRoutes {
                        t.Logf("Expected AttachedRoutes to be %v, got %v", eListener.AttachedRoutes, aListener.AttachedRoutes)
                        return false
                }
```

```golang
                if !conditionsMatch(t, eListener.Conditions, aListener.Conditions) {
                        t.Logf("Expected Conditions to be %v, got %v", eListener.Conditions, aListener.Conditions)
                        return false
                }
```

e.g.:

`go test -coverprofile cover.out ./conformance/utils/kubernetes/...`
`go tool cover -html=cover.out -o cover.html`
`firefox cover.html`

Reply:
yah, these should be tracked with a GH issue and fixed in a follow up PR :)

Reply:
If this is pushing it for your available time, then I can agree to making it a follow-up as what we have here would appear to be a significant improvement over the previously broken logic. Thank you for your time so far :bow:

Reply:
Added to https://github.com/kubernetes-sigs/gateway-api/issues/1708 :+1: